src/msw/menu.cpp

```diff
@@ -225,29 +225,30 @@ void wxMenu::UpdateAccel(wxMenuItem *item)
             return;
         }
 
-        // find the (new) accel for this item
+        // remove old accels
+        int n = wxNOT_FOUND;
+        while ( (n = FindAccel(item->GetId())) != wxNOT_FOUND )
+        {
+            delete m_accels[n];
+            m_accels.RemoveAt(n);
+        }
+
+        // find the (new) accel for this item and add it if any
         wxAcceleratorEntry *accel = wxAcceleratorEntry::Create(item->GetItemLabel());
         if ( accel )
-            accel->m_command = item->GetId();
-
-        // find the old one
-        int n = FindAccel(item->GetId());
-        if ( n == wxNOT_FOUND )
         {
-            // no old, add new if any
-            if ( accel )
-                m_accels.Add(accel);
-            else
-                return;     // skipping RebuildAccelTable() below
+            accel->m_command = item->GetId();
+            m_accels.Add(accel);
         }
-        else
+
+        // add extra accels
+        const wxVector<wxAcceleratorEntry>& extraAccelsVector = item->GetExtraAccels();
+        const int extraAccelsSize = extraAccelsVector.size();
```

Comment:
Doesn't this provoke a warning from MSVC about `size_t` to `int` conversion? We need to use `size_t` indices for the loops...